source/MRViewer/MRPlaneWidget.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include "MRMesh/MRBox.h"
+#include "MRMesh/MRPlane3.h"
+#include "MRViewer/MRViewer.h"
+
+namespace MR
+{
+// Visual widget to draw a plane
+// present in scene (ancillary), subscribes to viewer events
+class PlaneWidget : public MultiListener<MouseDownListener, MouseMoveListener, MouseUpListener>
+{
+    std::shared_ptr<ObjectMesh> planeObj_;
+    Plane3f plane_;
+    Box3f box_;
+    Vector3f cameraUp3Old_;
+    
+    using OnPlaneUpdateCallback = std::function<void()>;
+    OnPlaneUpdateCallback onPlaneUpdate_;
+
+    bool pressed_ = false;
+    Vector2f startMousePos_;
+    Vector2f endMousePos_;
+
+    std::shared_ptr<ObjectLines> line_;
+public:
+    
+    //updates plane, calls //returns plane if it is specified
+    MRVIEWER_API void updatePlane( const Plane3f& plane, bool updateCameraRotation = true );
+    //updates box
+    MRVIEWER_API void updateBox( const Box3f& box, bool updateCameraRotation = true );
```

Comment:
please write more meaningful comment about how box is used and what is updateCameraRotation for